mitodl/hugo-course-publisher
# set up hugo site for multi-course

#### Pre-Flight checklist

- [x] no style changes

#### What are the relevant tickets?

none

#### What's this PR do?

this copies over an updated-ish copy of the linear algebra course and the geology course and sets up the site for rendering multiple courses. basically:

- courses are moved into `site/content/courses/$COURSE_ID`
- in there there is a `_index.md` file and `resources` directory
- the templates are moved around to render these things properly

I also added a new template for `index.html` which is just a placeholder for now, and a layout at `site/layouts/courseindex/section.html` which ends up being rendered for `/courses`. this is basically just a simple list of the courses registered in the site right now.

The idea is that as we continue improving the pipeline for churning out markdown we can start copy the markdown for all the test courses over into this project, and so any thing we're testing as far as layout changes we can start testing on all the courses at once. This will also make it easy for us to push up new rendered versions of these for people to review (e.g. to github pages or what have you).

In the long run this may or may not be how we want to render these things.

note that to get this working I had to do a little editing of the markdown, to change the menu keys to be course-specific. this will be changed upstream in the ocw-to-hugo repo, and once that is set up I'll make the change over here as well. 

#### How should this be manually tested?

you should be able to pull down the branch and fire it up with no error. visiting `/courses` should give you a list of the included courses (just two right now) and, from there, you should be able to visiting them and click around. there should be no regressions or weirdness there.
